pkg/platform/runtime2/runtime.go

```diff
@@ -29,7 +29,11 @@ func IsNeedsUpdateError(err error) bool {
 	return errs.Matches(err, &NeedsUpdateError{})
 }
 
-func New(target setup.Targeter) (*Runtime, error) {
+func new(target setup.Targeter, noSetupYet bool) (*Runtime, error) {
+	if noSetupYet {
```

Comment:
I realized that `"runtime", "start"` also needs to be called during `runtime.New()` not only during the `Update()` method.  But unfortunately, the `runtime.New()` method is called by the `runtime.Update()` function as well.  And we do not want to double-send the event.  So, I had to add this (rather awkward) flag to check if we were called from the `Update()` function. :(

Reply:
This of course, could be removed if we had a smarter analytics call, that ensures that Events are called only once.

Reply:
:facepalm: :+1:

Reply:
@Naatan It is fixed.  And I suppose you want me to file a story similar to what I proposed here: 

> To do this cleanly (and not with lots of hard-to-grok) state variables and if-clauses inside the runtime package, we probably need a separate struct (maybe RuntimeAnalytics) that we instantiate somewhere at the runner level, and then can forward to runtime.New() and runtime.Update() calls. This struct could ensure that every Event is sent only once during an instances lifetime.

Reply:
Yes please, by all means file a story for that :)